ckanext/adfs/plugin.py

```diff
@@ -17,14 +18,9 @@
 X509 = ''
 WSFED_ENDPOINT = ''
 WTREALM = pylons.config['adfs_wtrealm']
-with open(pylons.config['adfs_federation_metadata_path']) as federation_meta:
-    METADATA = ET.fromstring(federation_meta.read())
-    for child in list(METADATA):
-        service_type = child.attrib.get('{http://www.w3.org/2001/XMLSchema-instance}type', None)
-        if service_type == 'fed:ApplicationServiceType':
-            X509 = child.find('{urn:oasis:names:tc:SAML:2.0:metadata}KeyDescriptor').xpath('string()')
-            WSFED_ENDPOINT = child.find('{http://docs.oasis-open.org/wsfed/federation/200706}PassiveRequestorEndpoint').xpath('string()')
-            break
+METADATA = get_federation_metadata(pylons.config['adfs_federation_metadata_url'])
```

Comment:
We are inescapably interacting with M$ to implement auth via the M$ Azure platform ? 

The trade-off is 1X extra network request per login (Speed) vs Login *breaking* if the cert changes. (UX).

The only way we would currently be notified about the cert changing is our login mechanism breaking & annoying someone enough that they raise a Zendesk ticket.

Consider mean response time for escalating such issues combined w. BAU support arrangements as they stand.

I vote for a slightly slower login w/out an obvious known failure mode :)